ui/src/app/components/baseComponent.ts

```diff
@@ -61,7 +61,7 @@ export abstract class PureComponent<P extends PureComponentProps, S extends Pure
         this.setMultiState(newState, callback);
     }
 
-    protected setMultiState(newState: any, callback?: () => void): void {
+    protected setMultiState<K extends keyof S>(newState: Pick<S, K>, callback?: () => void): void {
```

Comment:
OK what does this do?  :)

Reply:
it makes "forwards" the generics to the react set state function, so there is type checking at compilation time of the objects set in the state